platform-api/che-core-api-infrastructure-local/src/main/java/org/eclipse/che/api/local/LocalAccountDaoImpl.java

```diff
@@ -414,4 +423,55 @@ public SubscriptionQueryBuilder getSubscriptionQueryBuilder() {
         }
         return lockedAccounts;
     }
+
+    /** {@inheritDoc} */
+    @Override
+    public List<Account> find(AccountSearchCriteria searchCriteria, int page, int perPage) throws ServerException {
+        List<Account> result = new ArrayList<>(accounts);
+        try {
+            if (searchCriteria.getEmailOwner() != null) {
+                User owner = userDao.getByAlias(searchCriteria.getEmailOwner());
+                List<Account> byOwner = getByOwner(owner.getId());
+                merge(result, byOwner);
+            }
+
+            if (searchCriteria.getId() != null) {
+                Account account = getById(searchCriteria.getId());
+                merge(result, ImmutableList.of(account));
+            }
+
+            if (searchCriteria.getName() != null) {
+                Account account = getByName(searchCriteria.getName());
+                merge(result, ImmutableList.of(account));
+            }
+
+            if (searchCriteria.getSubscription() != null) {
+                List<Account> bySubscriptions = new LinkedList<>();
+                for (Subscription subscription : subscriptions) {
+                    if (subscription.getServiceId().equals(searchCriteria.getSubscription())) {
+                        bySubscriptions.add(getById(subscription.getAccountId()));
+                    }
+                }
+                merge(result, bySubscriptions);
+            }
+
+        } catch (NotFoundException e) {
```

Comment:
Try catch set is too wide. NotFound exception can be in getById(searchCriteria.getId()); and as rusult you will have empty list